src/main/kotlin/io/hsar/wh40k/combatsimulator/model/World.kt

```diff
@@ -1,13 +1,40 @@
 package io.hsar.wh40k.combatsimulator.model
 
 import io.hsar.wh40k.combatsimulator.logic.TurnAction
+import io.hsar.wh40k.combatsimulator.model.unit.BaseStat
+import kotlin.math.absoluteValue
+import kotlin.math.max
 
-data class World(val friendlyForces: List<UnitInstance>, val enemyForces: List<UnitInstance>) {
+data class World(val friendlyForces: List<UnitInstance>, val enemyForces: List<UnitInstance>,
+                 val unitPositions: Map<UnitInstance, MapPosition>) {
 
     fun executeActions(executingUnit: UnitInstance, actionsToExecute: List<TurnAction>) {
         actionsToExecute
                 .map { actionToExecute ->
                     executingUnit.executeTurnAction(actionToExecute)
                 }
     }
+
+    // Used by things like TacticalActionStrategy to work out how far away units are from each other
+    // This will give the inclusive distance from square to square, so callers need to subtract 1 when working out
+    // the distance they need to travel to be NEXT to a unit
+    fun distanceApart(unit: UnitInstance, otherUnit: UnitInstance): Int {
+        val unitPosition = unitPositions[unit]
+        val otherUnitPosition = unitPositions[otherUnit]
+        return otherUnitPosition?.let { unitPosition?.distanceToPosition(it)} ?: Int.MAX_VALUE
```

Comment:
I think that if there are null values you should `throw IllegalArgumentException("Some useful message")` rather than returning an arbitrary error value.

Reply:
Yep, that seems reasonable. Will change on feature branch.

Reply:
Changed to use .getValue()